tests/js/zamboni/stats.spec.js

```diff
@@ -0,0 +1,127 @@
+const dateMock = require('jest-date-mock');
+
+describe(__filename, () => {
+  const defaultBaseUrl = 'http://example.org/';
+
+  // This should be global to all stats files.
+  beforeEach(() => {
+    // Mock mandatory jQuery plugins.
+    $.prototype.modal = jest.fn();
+    $.prototype.csvTable = jest.fn();
+    $.prototype.datepicker = () => ({ datepicker: jest.fn() });
+    $.datepicker = { setDefaults: jest.fn() };
+
+    global.z = {
+      SessionStorage: jest.fn(),
+      Storage: jest.fn(),
+      capabilities: {},
+    };
+
+    global._pd = (func) => {
+      return function (e) {
+        e.preventDefault();
+        func.apply(this, arguments);
+      };
+    };
+  });
+
+  afterEach(() => {
+    dateMock.clear();
+  });
+
+  describe('stats/stats.js', () => {
+    const report = 'apps';
+
+    let stats_stats;
+
+    beforeEach(() => {
+      stats_stats = require('../../../static/js/zamboni/stats-all.js')
```

Comment:
Using the compiled (or just concatenated?) version is easier because of the various dependencies. Fortunately, all deps are not global. We could use the minified version but that makes reading errors tricky.

Reply:
It needs to be `require`'ed that way because of the few global mocks declared above.